pkg/registry/apis/provisioning/resources/dualwriter.go

```diff
@@ -35,17 +35,20 @@ func (r *DualReadWriter) Read(ctx context.Context, path string, ref string) (*Pa
 
 	info, err := r.repo.Read(ctx, path, ref)
 	if err != nil {
-		return nil, fmt.Errorf("read file: %w", err)
+		if _, ok := err.(*apierrors.StatusError); ok {
+			return nil, err
```

Comment:
This is the proper way to parse the error (lifted from https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L810-L812):
```suggestion
		if status, ok := err.(apierrors.APIStatus); ok || errors.As(err, &status) {
			return nil, status
```

Side-note, we might want to turn that bit of logic into a helper (e.g. in `pkg/apimachinery/utils/errors.go` for now, but eventually all this should live in app SDK) – I expect it to be used in plenty of places and working with `errors.As` is super unintuitive IMHO.

Reply:
Done